samples/reactjs-todo/client/components/journey/webAuthn.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * forgerock-sample-web-react
+ *
+ * webAuthn.js
+ *
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+import { FRWebAuthn, WebAuthnStepType } from '@forgerock/javascript-sdk';
+import React, { useEffect, useState } from 'react';
+import Loading from '../utilities/loading';
+
+/**
+ * @function WebAuthn - Footer React component
+ * @returns {Object} - React component object
+ */
+export default function WebAuthn({ step, setSubmissionStep }) {
+  const webAuthnStep = FRWebAuthn.getWebAuthnStepType(step);
+  const [state, setState] = useState({
+    message: '',
+    header: '',
+  });
+  useEffect(() => {
```

Comment:
I see we are using a functional setState in this effect. I'm curious why we made this distinction?

It doesn't seem like its possible to set the state multiple times before the render in this control flow? I think we should make it a regular setState unless i'm missing something to make it seem a bit less complex.

Functional setState comes with a little more complexity.

Reply:
you mean two different states like this,one for the header and another for message?. Going thru react docs i saw i could "combine" two values into one unique state and went for it. Is it a bad practice? I could use an explanation for upcoming features
<img width="553" alt="image" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/c0ad9fed-406a-4657-9062-ee80298c7a04">

Reply:
I am referring to this

   ```
         setState((state) => ({
            ...state,
            message: 'Your device will be used to verify your identity',
          }));
   ```
   
   why not
   
   `setState({...state,message: 'Your device will be used to verify your identity'})`

Reply:
i just tried and it enters into a loop. Reacts aks for the state to be added as a dependency and when i test it, it enters into a loop, trying to constantly log in. 
I assumed i was doing this to have the value for the previous state, like a reducer in redux pattern,
```
setState((state) => ({
         ...state,
         message: 'Your device will be used to verify your identity',
}))
```;